go/enclave/evm/evm_facade.go

```diff
@@ -268,25 +268,34 @@ func readVisibilityConfig(vmenv *vm.EVM, contractAddress *gethcommon.Address) *c
 		return &core.ContractVisibilityConfig{AutoConfig: true}
 	}
 
+	transp := false
+	if visibilityRules.ContractCfg == transparent {
+		transp = true
+	}
+
 	cfg := &core.ContractVisibilityConfig{
 		AutoConfig:   false,
-		Transparent:  &visibilityRules.IsTransparent,
+		Transparent:  &transp,
 		EventConfigs: make(map[gethcommon.Hash]*core.EventVisibilityConfig),
 	}
 
 	for i := range visibilityRules.EventLogConfigs {
 		logConfig := visibilityRules.EventLogConfigs[i]
-
-		sig := gethcommon.Hash{}
-		sig.SetBytes(logConfig.EventSignature)
-
-		cfg.EventConfigs[sig] = &core.EventVisibilityConfig{
+		relevantToMap := make(map[uint8]bool)
+		for _, field := range logConfig.VisibleTo {
+			relevantToMap[field] = true
+		}
+		t1 := relevantToMap[topic1]
+		t2 := relevantToMap[topic2]
+		t3 := relevantToMap[topic3]
+		s := relevantToMap[sender]
+		cfg.EventConfigs[logConfig.EventSignature] = &core.EventVisibilityConfig{
```

Comment:
Those really need a comment with examples to better understand event visibility.

Reply:
the comments are in the config. This is the implementation.

Aren't the comments in [ContractTransparencyConfig.sol](https://github.com/ten-protocol/go-ten/pull/2066/files/a93e2864ffa2c8398b65f760442da264a12ad284#diff-c78ef949c42159be405e9be2cc53c3d18c63da6041ba84f5fe9a37f6d5d7dd0b)enough?